src/search/search.go

```diff
@@ -44,21 +47,31 @@ func PerformSearch(query string, options engines.Options, config *config.Config)
 // engine_searcher, NewEngineStarter()  use this.
 type EngineSearch func(context.Context, string, *bucket.Relay, engines.Options, config.Settings) error
 
-func runEngines(engineMap map[string]config.Engine, query string, worker *conc.WaitGroup, relay *bucket.Relay, options engines.Options) {
+func runEngines(engs []engines.Name, settings map[string]config.Settings, query string, worker *conc.WaitGroup, relay *bucket.Relay, options engines.Options) {
+	config.EnabledEngines = engs
 	log.Info().Msgf("Enabled engines (%v): %v", len(config.EnabledEngines), config.EnabledEngines)
 
 	engineStarter := NewEngineStarter()
-	for name, engine := range engineMap {
-		engineName, nameErr := engines.NameString(name)
-		if nameErr != nil {
-			log.Panic().Err(nameErr).Msg("failed converting string to engine name")
-			return
-		}
-
+	for i := range engs {
+		eng := engs[i] // dont change for to `for _, eng := range engs {`, eng retains the same address throughout the whole loop
 		worker.Go(func() {
-			if err := engineStarter[engineName](context.Background(), query, relay, options, engine.Settings); err != nil {
-				log.Error().Err(err).Msgf("failed searching %v", engineName)
+			strt := engineStarter[eng]
+			err := strt(context.Background(), query, relay, options, settings[eng.ToLower()])
```

Comment:
todo: create context with deadlines for each engine run https://github.com/tminaorg/brzaguza/issues/115